submissions/AriadnaKravchuk/memory-game/script.js

```diff
@@ -0,0 +1,179 @@
+
+        if (secondsCounter === 59) {
+            secondsCounter = 0;
+            minutesCounter++;
+        }
+
+        gameTime.innerHTML = `Time: ${minutes}.${seconds}`;
+        setTimeout(countTime, TIMEOUT.TIMER);
+    }
+}
+
+function addCardsImages() {
+    let frontImage = theme === "gothic" ? "※" : "★";
+
+    for (let i = 0; i < CARD_IMG.QUANTITY * 2; i++) {
+        cardsFronts[i].innerHTML = frontImage;
+        cardsIcons[i].src =
+            CARD_IMG.SOURSE + theme + "/" + cardsIcons[i].id + CARD_IMG.TYPE;
+    }
+}
+
+function changeCardsOrders() {
+    cards.forEach((card) => {
+        card.style.order = Math.floor(Math.random() * 16);
+    });
+}
+
+function changeTheme(event) {
+    body.className = event.target.id;
+    theme = event.target.id;
+    gamePage.classList.remove("hidden");
+    startPage.classList.add("hidden");
+    allDetails.forEach((details) => {
+        details.open = false;
+    });
+    addCardsImages();
+    changeCardsOrders();
+}
+
+function returnMenu() {
+    startPage.classList.remove("hidden");
+    gamePage.classList.add("hidden");
+    popup.classList.add("hidden");
+    cardsFronts.innerHTML = "";
+    cardsInner.forEach((card) => {
+        card.classList.remove("flipped");
+        card.addEventListener("click", flipCard);
+    });
+}
+
+function restartGame() {
+    cardsInner.forEach((card) => {
+        card.classList.remove("flipped");
+        card.addEventListener("click", flipCard);
+        setTimeout(() => {
+            changeCardsOrders();
+            popup.classList.add("hidden");
+        }, TIMEOUT.CARD_UNFLIPPED);
+    });
+}
+
+function finishGame(event) {
+    event.target.id === "menu" ? returnMenu() : restartGame();
+
+    matches = 0, moves = 0, secondsCounter = 0, minutesCounter = 0;
+    (isEndGame = false), (isStartGame = true);
+    gameTime.innerHTML = `Time: 00.00`;
+    gameMoves.innerHTML = `Moves: 0`;
+}
+
+function flipCard() {
+    if (isStartGame) {
+        setTimeout(countTime, TIMEOUT.TIMER);
+        isStartGame = false;
+    }
+
+    if (isLock) return;
+    if (this === firstCard) return;
+
+    this.classList.add("flipped");
+
+    if (!isFlipped) {
+        isFlipped = true;
+        firstCard = this;
+    } else {
+        isFlipped = false;
+        secondCard = this;
+
+        moves++;
+        gameMoves.innerHTML = `Moves: ${moves}`;
+
+        checkMatches();
+    }
+}
+
+function checkMatches() {
+    if (firstCard.id === secondCard.id) {
+        matches++;
+        firstCard.removeEventListener("click", flipCard);
+        secondCard.removeEventListener("click", flipCard);
+
+        if (matches === CARD_IMG.QUANTITY) {
+            isEndGame = true;
+            scorePopup.innerHTML = `Time: ${minutes}.${seconds} Moves: ${moves}`;
+            setTimeout(() => {
+                popup.classList.remove("hidden");
+            }, TIMEOUT.SHOW_POPUP);
+        }
+        firstCard = [], secondCard = [];
```

Comment:
Same.  Why are you using `array`? It is truly value and you don't use array or its methods. Maybe `null` or `undefined` will be better choose.